### DISCUSSION TITLE
Camera stream as texture

### DISCUSSION THREAD

**Discussion by adrianposteuca** [google/model-viewer#4532] [Q&A]
Hi all ,

I would like to implement an idea. I want to use the camera video stream as a videoTexture.

my approach was to use this type of code 

`customElements.whenDefined('model-viewer').then(async () => {
  const videoElement = document.querySelector("#cameraFeed");
  
  try {
    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
    videoElement.srcObject = stream;
    await new Promise(resolve => videoElement.onplaying = resolve);
    videoTexture = modelViewer.createVideoTexture(videoElement);
  } catch (err) {
    console.error("Error accessing the camera:", err);
  }
});`
 it doesn't seem to work

Any ideas?

**Comment by elalish** [comment]
I have no idea if that's even possible, but if you get it working, please let us know!

**Reply by adrianposteuca** [reply]
Something like this but on web https://youtu.be/-oxjG9mg6Os
